src/modules/Visualizer/Story/Story.js

```diff
@@ -20,123 +21,108 @@ const resetLayers = map => ({ layers = [] }) => {
   });
 };
 
-export class Story extends React.Component {
-  static propTypes = {
-    story: PropTypes.shape({
-      beforeEach: PropTypes.arrayOf(PropTypes.shape({
-        layers: PropTypes.arrayOf(PropTypes.string),
-        active: PropTypes.bool,
-      })),
-      slides: PropTypes.arrayOf(PropTypes.shape({
-        title: PropTypes.string,
-        content: PropTypes.string,
-        layouts: PropTypes.arrayOf(PropTypes.shape({
-          layers: PropTypes.arrayOf(PropTypes.string),
-          active: PropTypes.bool,
-        })),
-        legends: PropTypes.arrayOf(PropTypes.shape({
-          title: PropTypes.string,
-          items: PropTypes.arrayOf(PropTypes.shape({
-            label: PropTypes.string,
-            color: PropTypes.string,
-          })),
-        })),
-      })),
-    }).isRequired,
-  }
-
-  state = {
-    step: 0,
-  }
+export const Story = ({ map, story: { beforeEach = [], slides }, setLegends, translate }) => {
+  const [step, setStep] = useState(0);
 
-  componentDidMount () {
-    this.initMap();
-  }
+  const slidesLength = slides.length - 1;
```

Comment:
```suggestion
  const slidesLength = React.useMemo(() => (slides.length - 1), [slides.length]);
```

Reply:
With your suggestion, we allocate another function with `slides.length` as argument to return... `slides.length - 1`.
It's not a huge calculation and i am not sure if it's better for performance purposes.
What do you think @jrmi ?

Reply:
This insure that `slidesLength` won't change between renders when slides do not change…

But ultimately, as it is just a Number, it does not really matter.

So yep, you can forget about that ;)

Reply:
No need useMemo here, a number don't change is reference ! The useMemo is only usefull for high computation which is no the case here.

Reply:
My problem is more the naming here. Are we really talking about length of a silde ? I propose `slideCount` instead.
```suggestion
  const slideCount = slides.length - 1;
```